shell/platform/windows/angle_surface_manager.cc

```diff
@@ -300,6 +305,24 @@ EGLSurface AngleSurfaceManager::CreateSurfaceFromHandle(
                                           egl_config_, attributes);
 }
 
+void AngleSurfaceManager::SetVSyncEnabled(bool enabled) {
+  if (eglMakeCurrent(egl_display_, render_surface_, render_surface_,
+                     egl_context_) != EGL_TRUE) {
+    LogEglError("Unable to make surface current to update the swap interval");
+    return;
+  }
+
+  // OpenGL swap intervals can be used to prevent screen tearing.
+  // If enabled, the raster thread blocks until the v-blank.
```

Comment:
Does this mean that we will still block until the v-blank when this parameter is true, i.e. when composition is not enabled? And if so, is that what we intend?

Reply:
Yes and yes. Screen tearing is possible if a Windows 7 user does not have DWM composition enabled. The app should synchronize with the vsync to prevent screen tearing.